usr.sbin/daemon/daemon.c

```diff
@@ -534,14 +535,14 @@ daemon_sleep(time_t secs, long nsecs)
 
 static void
 open_pid_files(const char *pidfile, const char *ppidfile,
-	       struct pidfh **pfh, struct pidfh **ppfh)
+	       struct daemon_state * state)
 {
 	pid_t fpid;
 	int serrno;
 
 	if (pidfile) {
-		*pfh = pidfile_open(pidfile, 0600, &fpid);
-		if (*pfh == NULL) {
+		state->child_pidfh = pidfile_open(pidfile, 0600, &fpid);
+		if (state->child_pidfh == NULL) {
 			if (errno == EEXIST) {
 				errx(3, "process already running, pid: %d",
 				    fpid);
```

Comment:
Why does this exit path not call the common exit path code?
A comment about why not would be good to have here.
Though it looks to be only to generate better error messages (which is good) and different return codes which are documented, so a comment here would clear that up... and a quick back reference below...

Reply:
The change at hand does not touch the logic of the function, I was simply renaming and relocating the variable. So my answer to the question "Why does this exit path not call the common exit path code?" is I don't know. But I will take this as a TODO to clean this up or clarify with comments

Reply:
That makes sense. I'm happy to defer this to a TODO for future enhancement.